include/libdnf5/repo/repo.hpp

```diff
@@ -387,21 +383,17 @@ class Repo {
     /// @return Whether at least the repodata cache cloning was successful.
     bool clone_root_metadata();
 
-    libdnf5::BaseWeakPtr base;
-    ConfigRepo config;
+    RepoDownloader & get_downloader() const;
```

Comment:
What about moving everything from the private part into the pimpl?

Reply:
I mean, if we have already decided that a class is a candidate for potential future changes, it's worth mitigating any risks by keeping the exposed set of methods and members in public header files to a minimum.

Reply:
After the discussion, private members here are not used in any inline function, so it should be safe to potentially remove them in future.